containers/daap-python-base/src/var/task/data_platform_paths.py

```diff
@@ -198,25 +199,51 @@ def extract_timestamp_from_curated_path(string: str):
     return search_string_for_regex(string, regex=LOAD_TIMESTAMP_CURATED_REGEX)
 
 
-def get_latest_version(data_product_name: str):
-    """Get the latest version of a data product, else return v1.0"""
+def get_all_versions(data_product_name: str) -> list[str]:
+    """Returns all versions of a given data product"""
     metadata_bucket = get_metadata_bucket()
-    metadata_versions = s3.list_objects_v2(
+    metadata_versions = s3_client.list_objects_v2(
         Bucket=metadata_bucket, Prefix=data_product_name + "/"
     )
     # This is the case in which the data product is new.
     if not metadata_versions.get("Contents"):
-        return "v1.0"
+        return ["v1.0"]
 
     versions = [
         version["Key"].split("/")[1]
         for version in metadata_versions["Contents"]
         if version["Size"] > 0
     ]
     versions.sort(key=lambda x: [int(y.replace("v", "")) for y in x.split(".")])
-    latest_version = versions[-1]
 
-    return latest_version
+    return versions
+
+
+def get_latest_version(data_product_name: str) -> str:
+    return get_all_versions(data_product_name=data_product_name)[-1]
+
+
+def generate_all_element_version_prefixes(
+    path_prefix: str, data_product_name: str, table_name: str
+) -> list[str]:
+    """Generates element prefixes for all data product versions"""
+
+    data_product_versions = get_all_versions(data_product_name)
+    element_prefixes = []
+
+    for version in data_product_versions:
+        element_prefixes.append(
+            f"{path_prefix}/{data_product_name}/{version}/{table_name}/"
+        )
+
+    return element_prefixes
+
+
+def s3_recursive_delete(bucket_name: str, prefixes: list[str]) -> None:
```

Comment:
You think this would be better in versioning as well?

Reply:
Agreed, now moved to the versioning module.